great_expectations/experimental/datasources/sql_datasource.py

```diff
@@ -86,50 +88,47 @@ class SqlYearMonthSplitter(ColumnSplitter):
         default_factory=lambda: ["year", "month"]
     )
 
-    def param_defaults(self, data_asset: DataAsset) -> Dict[str, List]:
-        """Query the database to get the years and months to split over.
+    def param_defaults(self, table_asset: TableAsset) -> Dict[str, List]:
+        """Query sql database to get the years and months to split over.
 
         Args:
-            data_asset: A TableAsset over which we want to split the data.
+            table_asset: A TableAsset over which we want to split the data.
         """
-        # This column splitter is only relevant to SQL data assets so we do some assertions
-        # to validate this.
-        from great_expectations.execution_engine import SqlAlchemyExecutionEngine
+        return _query_for_year_and_month(
+            table_asset, self.column_name, _get_sql_datetime_range
+        )
+
 
-        assert isinstance(data_asset, TableAsset), "data_asset must be a TableAsset"
-        assert isinstance(
-            data_asset.datasource.execution_engine, SqlAlchemyExecutionEngine
+def _query_for_year_and_month(
```

Comment:
The strategy here is to take the code that use to be in `param_defaults` and put it here. We parameterize the `query_datetime_range` `Callable` which let's us break out the sqlite specific code.